routers/web/repo/pull.go

```diff
@@ -587,7 +587,7 @@ func PrepareViewPullInfo(ctx *context.Context, issue *issues_model.Issue) *git.C
 	ctx.Data["HeadBranchCommitID"] = headBranchSha
 	ctx.Data["PullHeadCommitID"] = sha
 
-	if pull.HeadRepo == nil || !headBranchExist || headBranchSha != sha {
+	if pull.HeadRepo == nil || !headBranchExist {
```

Comment:
We've run into cases where a PR had a difference in these SHAs and so was shown as broken, even when it had never been closed. But we have not been able to get a clear repro case yet.

Maybe this test can be skipped for closed PRs, but remain for open PRs? If what's shown in the PR and what's in the head branch is not the same, I think it's important to show that.

Reply:
okey, now skip the test when only the PR is closed